### ISSUE TITLE
RFE: Containerized build whose primary output artifact is a repo-bundle. 

### ISSUE THREAD

**Issue by timothysc** [vmware-tanzu/tanzu-framework#862]
**Describe the feature request**
As of today the primary build mechanics for the framework repo is a native gobuild.  The purpose of this issue is to pivot the build system to output as a set of OCI (container) artifacts that leverage container based build pipelines, whose primary output artifact is a repo bundle, or a consumable BOM fragment.  

This will simplify the downstream consumption of this fragment into a other distributions, such as TCE. 

**Affected product area (please put an X in all that apply)**

- [X] APIs
- [X] Addons
- [ X] CLI
- [ ] Docs
- [X] Installation
- [X] Plugin
- [ ] Security
- [X] Test and Release
- [X] User Experience

cc @joshrosso @zjs @vijaykatam @vuil @ichandrabhatta @figo

**Comment by timothysc** [comment]
Yeah I think this one is the most comprehensive, I'd move to close the others in preference for this one.

**Comment by saji-pivotal** [comment]
Since we've identified there are dupes of this issue, I'll remove the needs-triage label from this
